pkg/scheduler/partition.go

```diff
@@ -79,7 +79,29 @@ type PartitionContext struct {
 	locking.RWMutex
 }
 
+// newPartitionContextForValidation initializes a shadow partition based on the configuration.
+// The shadow partition is used to validate the configuration, it is not used for scheduling.
+func newPartitionContextForValidation(conf configs.PartitionConfig, rmID string, cc *ClusterContext) (*PartitionContext, error) {
```

Comment:
The return value `*PartitionContext` is never used from this function, so you might as well just remove it and just call it `validateConfiguration()`. Unless of course, you create unit tests which actually use it for some kind of verification... See other comments.